awx/main/managers.py

```diff
@@ -127,11 +130,21 @@ def register(self, uuid=None, hostname=None, ip_address=None, node_type='hybrid'
                         other_inst.save(update_fields=['ip_address'])
                         logger.warning("IP address {0} conflict detected, ip address unset for host {1}.".format(ip_address, other_hostname))
 
-            # Return existing instance that matches hostname
+            # get the instance based on the hostname
             instance = self.filter(hostname=hostname)
+
+            # Check or update the existing uuid
+            if uuid != None and uuid != self.UUID_DEFAULT:
```

Comment:
this should be written as `uuid is not None...`

Reply:
I kept it this way to match the existing if statements below, but I'm not set on that.

Reply:
i conditionally approved, pending these changes!

Reply:
updated, cheers y'all